SmartDeviceLink-iOS/SmartDeviceLink/SDLProtocolHeader.m

```diff
@@ -38,20 +38,27 @@ - (NSString *)description {
     return description;
 }
 
-
 + (SDLProtocolHeader *)headerForVersion:(UInt8)version {
-    if (version == 1) {
-        return [[SDLV1ProtocolHeader alloc] init];
-    }
-
-    if (version == 2) {
-        return [[SDLV2ProtocolHeader alloc] init];
-    }
-
     // TODO: some error handling here if unknown version is asked for,
     // but that needs to be balanced against future proofing. i.e. V3.
     // Requirements around V3 are as yet undefined so give a V2 header
-    return [[SDLV2ProtocolHeader alloc] init];
+    // lol wat.
+    switch (version) {
+        case 0: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:2];
+        } break;
+        case 1: {
+            return [[SDLV1ProtocolHeader alloc] init];
+        } break;
+        case 2:
+        case 3:
+        case 4: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:version];
+        } break;
+        default: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:2];
+        } break;
+    }
 }
```

Comment:
Regarding version 0, I believe there was a case where it might be sent, I'll have to ask @justinjdickow to see if he remembers why we left that. It may have been just because we weren't sure, and the old code did the same. The default case was possibly the same reason. We can reevaluate that, because I'd definitely prefer to fail faster than to leave corner cases.